sdks/java/io/hadoop-file-system/src/main/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystem.java

```diff
@@ -78,6 +79,11 @@
     for (String spec : specs) {
       try {
         FileStatus[] fileStatuses = fileSystem.globStatus(new Path(spec));
+        if (fileStatuses == null) {
+          return resultsBuilder.add(MatchResult.create(Status.NOT_FOUND,
```

Comment:
I don't think you want to return here yet, we should look up the rest of the results requested in the specs list.